### ISSUE TITLE
[Bug]: Incorrect calculation in Typography Tokens

### ISSUE THREAD

**Issue by Tayebsed93** [Orange-OpenSource/ouds-flutter#272]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

Iphone 14 Pro Max

### OS Version

iOS 18

### App version

v0.5.0

### Describe the issue

There is an issue in the Design System where the lineHeight is incorrectly calculated for typography tokens.

The current formula used is:

```
lineHeight = Line / Size
Example: 48 / 40 = 1.2
```

This leads to inaccurate values across all typography tokens.

Documentation Flutter : https://api.flutter.dev/flutter/painting/TextStyle/height.html

### Expected Behavior

The correct formula should be:

```
lineHeight = Size / Line
Example: 40 / 48 = 0.833
```

### Example:
For the DisplayLarge token:

- Font Size: 40px
- Line Height: 48px

✅ Expected lineHeight value:

**Comment by Tayebsed93** [comment]
📣 New Firebase App Distribution ALPHA upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (311)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by Tayebsed93** [comment]
📣 New TestFlight ALPHA upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (311)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by jerome-regnier** [comment]
@Tayebsed93 , Je viens de charger la version 0.5.0 Alpha sur Android. Impossible sur iOS qui rencontre un problème de téléchargement du package.
Toutefois, je te confirme que les textes sont bien trop proches voir, ils se chevauchent.
Je pense que ton calcul était le bon et que le ratio est inversé dans la documentation.

**Comment by Tayebsed93** [comment]
@jerome-regnier j'ai remis les valeurs comme avant. 
Les textes sont affichés sur 2,3 lignes pour tes tests

**Comment by Tayebsed93** [comment]
📣 New Firebase App Distribution ALPHA upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (313)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by Tayebsed93** [comment]
📣 New TestFlight ALPHA upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (313)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by jerome-regnier** [comment]
Hello @Tayebsed93 ,
Voici le test sur la version iOS :
https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=69226-3199&t=g0n4YMOTSk8LHmbx-1

J'aimerais qu'on échange peut etre en live pour caler ça car tout fonctionne bien pour le texte Display mais dès qu'on passe sur du Heading, Body et Label, tout décroche en letter spacing. Mais je me demande du coup si c'est la bonne police de caractère pour chaque style ?

**Comment by Tayebsed93** [comment]
hello @jerome-regnier suite à notre échange tu trouveras une nouvelle version : 

1. Remplacer la font ios : SFPro Display par SFPro
2. Changer le letter spacing sur Android avec le calcul suivant : `letterSpacing = letterSpacing / size`

**Comment by Tayebsed93** [comment]
📣 New Firebase App Distribution ALPHA upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (320)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by Tayebsed93** [comment]
📣 New TestFlight ALPHA upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (320)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by jerome-regnier** [comment]
@Tayebsed93 
Test iOS : https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=69231-80&t=ggvZPcwbcOMWfP0d-1
On a un problème de letter spacing

**Comment by jerome-regnier** [comment]
@Tayebsed93 
Test Android : https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=69232-115&t=cl6JmcnJOCaZYqyw-1
Idem que sur iOS, on a toujours un problème de letter spacing

**Comment by Tayebsed93** [comment]
@jerome-regnier comme convenu j'ai remplacé la font custom (Roboto/SfPro) par la font du system

**Comment by Tayebsed93** [comment]
📣 New Firebase App Distribution ALPHA upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (322)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by Tayebsed93** [comment]
📣 New TestFlight ALPHA upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (322)
Version: 0.5.0
Build type: ALPHA
Build details (GitHub): 272
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @MaximeTonnerre @jerome-regnier (design team)

**Comment by jerome-regnier** [comment]
Test sur Android :
https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=69232-115&t=jSnlRAzCstuOHCDa-1

Test sur iOS :
https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=69231-80&t=jSnlRAzCstuOHCDa-1

On a toujours le letter spacing qui ne correspond pas sur certaines tailles de police.

**Comment by Tayebsed93** [comment]
I'm marking the issue as blocking !
For now, there's no solution. Investigation will continue after @jerome-regnier returns from vacation.

**Comment by jerome-regnier** [comment]
Design Review Figma
https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=71016-22700&t=sCvuoZFLsF3x3IMa-1  
Il semble y avoir un problème de graisse sur la font sur tous les style utilisant Moderate

**Comment by jerome-regnier** [comment]
Design Review Figma :
https://www.figma.com/design/yKtkEY1DOl7YDvmt7jq0QF/-OUDS-Core-Lib--Review?node-id=71142-20316&t=6LUNFZ2QrQOEZptH-1
👌 Ok Design pour la font Helvetica Neue
